src/main/java/moze_intel/projecte/gameObjs/entity/EntityHomingArrow.java

```diff
@@ -1,105 +1,193 @@
+import javax.vecmath.Matrix4f;
+import javax.vecmath.Quat4f;
+import javax.vecmath.Vector3d;
+import java.util.Collections;
+import java.util.Comparator;
 import java.util.List;
 
 public class EntityHomingArrow extends EntityArrow
 {
-	EntityLiving target;
-	World world;
+	private static final int DW_TARGET_ID = 31;
+	private static final int NO_TARGET = -1;
 
-	private void init(World world)
-	{
-		this.world = world;
-	}
+	private int newTargetCooldown = 0;
 
 	public EntityHomingArrow(World world)
 	{
 		super(world);
-		init(world);
 	}
 
 	public EntityHomingArrow(World world, EntityLivingBase par2, float par3) 
 	{
 		super(world, par2, par3);
-		init(world);
 	}
-	
+
 	@Override
-	public void onUpdate()
+	public void entityInit()
 	{
-		//TODO Create proper custom arrow. This one is duplicating because of the super call for onUpdate();
-		super.onUpdate();
+		super.entityInit();
+		dataWatcher.addObject(DW_TARGET_ID, NO_TARGET); // Target entity id
+	}
 
-		AxisAlignedBB box = this.boundingBox;
-		
-		if (target == null && !WorldHelper.isArrowInGround(this))
+	@Override
+	public void onUpdate()
+	{
+		onEntityUpdate();
+		boolean inGround = WorldHelper.isArrowInGround(this);
+		if (!worldObj.isRemote)
 		{
-			AxisAlignedBB bBox = box.expand(8, 8, 8);
-			List<EntityLiving> list = this.worldObj.getEntitiesWithinAABB(EntityLiving.class, bBox);
-
-			double distance = 100000;
+			if (hasTarget() && (!getTarget().isEntityAlive() || inGround))
+			{
+				dataWatcher.updateObject(DW_TARGET_ID, NO_TARGET);
+			}
 
-			for (EntityLiving entity : list)
+			if (!hasTarget() && !inGround && newTargetCooldown <= 0)
+			{
+				findNewTarget();
+			} else
 			{
-				double toIt = distanceTo(entity);
+				newTargetCooldown--;
+			}
+		}
 
-				if (distance > toIt)
+		if (ticksExisted > 3 && hasTarget() && !WorldHelper.isArrowInGround(this))
+		{
+			this.worldObj.spawnParticle("flame", this.posX + this.motionX / 4.0D, this.posY + this.motionY / 4.0D, this.posZ + this.motionZ / 4.0D, -this.motionX / 2, -this.motionY / 2 + 0.2D, -this.motionZ / 2);
+			this.worldObj.spawnParticle("flame", this.posX + this.motionX / 4.0D, this.posY + this.motionY / 4.0D, this.posZ + this.motionZ / 4.0D, -this.motionX / 2, -this.motionY / 2 + 0.2D, -this.motionZ / 2);
+			Entity target = getTarget();
+
+
+			Vector3d arrowLoc = new Vector3d(posX, posY, posZ);
+			Vector3d targetLoc = new Vector3d(target.posX, target.boundingBox.minY + target.height, target.posZ);
+
+			// Get the vector that points straight from the arrow to the target
+			Vector3d lookVec = new Vector3d(targetLoc);
+			lookVec.sub(arrowLoc);
+
+			// Find the angle between the direct vec and arrow vec, and then clamp it so it arcs a bit
+			double theta = wrap180Radian(arrowLoc.angle(lookVec));
+			theta = clampAbs(theta, 4 * Math.PI / 8);
+
+			// Find the cross product to determine the axis of rotation
+			Vector3d crossProduct = new Vector3d();
+			crossProduct.cross(arrowLoc, targetLoc);
```

Comment:
Same here:the rotation plane should be created by the heading and the desired heading